# Question
find correlation between attrition and other variables

In [35]:
#importing the required packages
import pandas as pd
from scipy.stats import pearsonr as ps
import matplotlib.pyplot as plt

In [77]:
d1=pd.read_csv('general_data.csv')
d=d1.dropna()
d.head()

Age  Attrition     BusinessTravel              Department  \
0   51          0      Travel_Rarely                   Sales   
1   31          1  Travel_Frequently  Research & Development   
2   32          0  Travel_Frequently  Research & Development   
3   38          0          0n-Travel  Research & Development   
4   32          0      Travel_Rarely  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                 6          2  Life Sciences              1           1   
1                10          1  Life Sciences              1           2   
2                17          4          Other              1           3   
3                 2          5  Life Sciences              1           4   
4                10          1        Medical              1           5   

   Gender          ...           NumCompaniesWorked Over18 PercentSalaryHike  \
0  Female          ...                          1.0      Y                11   
1  Female          ...                          0.0      Y                23   
2    Male          ...                          1.0      Y                15   
3    Male          ...                          3.0      Y                11   
4    Male          ...                          4.0      Y                12   

   StandardHours  StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0              8                 0               1.0                      6   
1              8                 1               6.0                      3   
2              8                 3               5.0                      2   
3              8                 3              13.0                      5   
4              8                 2               9.0                      2   

   YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0               1                        0                     0  
1               5                        1                     4  
2               5                        0                     3  
3               8                        7                     5  
4               6                        0                     4  

[5 rows x 24 columns]

# 1) 
 H0 – There is no significant difference in between the Attrition and Age of employee  
 H1 - There is significant difference in between  the Attrition and Age of employee 

In [65]:
s,p=ps(d.Attrition,d.Age)
print(s,p)

-0.15839867954096706 5.1265982193975044e-26


correlation r is -0.1583 and p is <0.05 therefor H0 is rejected



# 2)
H0 – There is no significant difference in between the Attrition and JobLevel of employee

H1 - There is significant difference in between the Attrition and JobLevel of employee


In [67]:
s,p=ps(d.Attrition,d.JobLevel)
print(s,p)

-0.012381569720790839 0.4125489150411943


correlation r is -0.0123815 and p is 0.412548 which is >=0.05 therefor H1 is rejected

# 3)
H0 – There is no significant difference in between the Attrition and MonthlyIncome	 of employee
 
H1 - There is significant difference in between the Attrition and MonthlyIncome	 of employee

In [68]:
s,p=ps(d.Attrition,d.MonthlyIncome)
print(s,p)

-0.030160293808460664 0.04589086274490169


correlation r is -0.03016 and p is 0.045 which is <0.05 therefor H0 is rejected

# 4)
H0 – There is no significant difference in between the Attrition and NumCompaniesWorked of employee

H1 - There is significant difference in between the Attrition and NumCompaniesWorked of employee

In [69]:
s,p=ps(d.Attrition,d.NumCompaniesWorked)
print(s,p)

0.04283056724472089 0.004572057121640626


correlation r is 0.04283 and p is 0.00457 which is <0.05 therefor H0 is rejected

# 5) 
H0 – There is no significant difference in between the Attrition and YearsSinceLastPromotion of employee

H1 - There is significant difference in between the Attrition and YearsSinceLastPromotion of employee

In [70]:
s,p=ps(d.Attrition,d.YearsSinceLastPromotion)
print(s,p)

-0.031423150563309944 0.037522936074080596


correlation r is -0.03142 and p is 0.03752 which is <0.05 therefor H0 is rejected

# 6) 
H0 – There is no significant difference in between the Attrition and Education of employee

H1 - There is significant difference in between the Attrition and Education	of employee

In [71]:
s,p=ps(d.Attrition,d.Education)
print(s,p)

-0.01710630705027875 0.25757539308237426


correlation r is -0.0171 and p is 0.25757 which is >=0.05 therefor H1 is rejected

# 7)
H0 – There is no significant difference in between the Attrition and YearsWithCurrManager of employee

H1 - There is significant difference in between the Attrition and YearsWithCurrManager of employee

In [72]:
s,p=ps(d.Attrition,d.YearsWithCurrManager)
print(s,p)

-0.15469153690287274 7.105369646795425e-25


correlation r is -0.154691 and p is  <0.05 therefor H0 is rejected

# 8)
H0 – There is no significant difference in between the Attrition and YearsAtCompany of employee

H1 - There is significant difference in between the Attrition and YearsAtCompany of employee

In [73]:
s,p=ps(d.Attrition,d.YearsAtCompany)
print(s,p)

-0.13300261842521538 9.47611808487322e-19


correlation r is -0.13300 and p is <0.05 therefor H0 is rejected

# 9)
H0 – There is no significant difference in between the Attrition and PercentSalaryHike of employee

H1 - There is significant difference in between the Attrition and PercentSalaryHike of employee

In [74]:
s,p=ps(d.Attrition,d.PercentSalaryHike)
print(s,p)

0.03315303713546665 0.02819244693520343


correlation r is 0.0331530 and p is 0.02819 which is <0.05 therefor H0 is rejected

# 10)
H0 – There is no significant difference in between the Attrition and DistanceFromHome of employee

H1 - There is significant difference in between the Attrition and DistanceFromHome of employee

In [75]:
s,p=ps(d.Attrition,d.DistanceFromHome)
print(s,p)

-0.00944863851515625 0.5317715668055768


correlation r is -0.009448 and p is 0.531771 which is >=0.05 therefor H1 is rejected

Correlation matrix for cross verification of correlation

In [76]:
d.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.158399          0.007376  -0.033900   
Attrition               -0.158399   1.000000         -0.009449  -0.017106   
DistanceFromHome         0.007376  -0.009449          1.000000  -0.007491   
Education               -0.033900  -0.017106         -0.007491   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008105  -0.004621         -0.000326  -0.009389   
JobLevel                -0.001137  -0.012382         -0.039990   0.045822   
MonthlyIncome           -0.045163  -0.030160         -0.022757   0.007289   
NumCompaniesWorked       0.299527   0.042831         -0.014449  -0.016210   
PercentSalaryHike       -0.032561   0.033153          0.037720  -0.041054   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031504  -0.008164          0.009353   0.002386   
TotalWorkingYears        0.680037  -0.169670          0.009574  -0.009228   
TrainingTimesLastYear   -0.028962  -0.047586         -0.008957   0.009939   
YearsAtCompany           0.311281  -0.133003          0.030746   0.005997   
YearsSinceLastPromotion  0.215650  -0.031423          0.002243   0.023457   
YearsWithCurrManager     0.201580  -0.154692          0.021773   0.005645   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008105 -0.001137      -0.045163   
Attrition                          NaN   -0.004621 -0.012382      -0.030160   
DistanceFromHome                   NaN   -0.000326 -0.039990      -0.022757   
Education                          NaN   -0.009389  0.045822       0.007289   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003090       0.007865   
JobLevel                           NaN   -0.003090  1.000000       0.046688   
MonthlyIncome                      NaN    0.007865  0.046688       1.000000   
NumCompaniesWorked                 NaN    0.000719 -0.009759      -0.021446   
PercentSalaryHike                  NaN   -0.004877  0.010874       0.004607   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.013488  0.000365       0.027242   
TotalWorkingYears                  NaN   -0.001688 -0.036293      -0.034398   
TrainingTimesLastYear              NaN   -0.012102 -0.031931       0.049780   
YearsAtCompany                     NaN    0.004117 -0.063360       0.000088   
YearsSinceLastPromotion            NaN    0.000814 -0.059680       0.064470   
YearsWithCurrManager               NaN    0.009079 -0.053898       0.023095   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299527          -0.032561            NaN   
Attrition                          0.042831           0.033153            NaN   
DistanceFromHome                  -0.014449           0.037720            NaN   
Education                         -0.016210          -0.041054            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                         0.000719          -0.004877            NaN   
JobLevel                          -0.009759           0.010874            NaN   
MonthlyIncome                     -0.021446           0.004607            NaN   
NumCompaniesWorked                 1.000000           0.030064            NaN   
PercentSalaryHike                  0.030064           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.016291           0.012104            NaN   
TotalWorkingYears                  0.238807          -0.018480            NaN   
TrainingTimesLastYear             -0.031335          -0.036720            NaN   
Years